src/js/drawer.js

```diff
@@ -0,0 +1,139 @@
+	let timeout;
+
+	const handleKeydown = (e) => {
+		if (e.keyCode === 27) {
+			callback();
+		}
+	};
+
+	const handleClick = (e) => {
+		if (!scope.contains(e.target)) {
+			console.log('callbacking');
+			callback();
+		}
+	};
+
+	const handleMouseenter = () => {
+		clearTimeout(timeout);
+	};
+
+	const handleMouseleave = () => {
+		timeout = setTimeout(callback, INTENT_DELAY);
+	};
+
+	const removeEvents = () => {
+		console.log('removing events');
+		clearTimeout(timeout);
+
+		scope.removeEventListener('mouseenter', handleMouseenter);
+		scope.removeEventListener('mouseleave', handleMouseleave);
+		document.removeEventListener('click', handleClick);
+		document.removeEventListener('touchstart', handleClick);
+		document.removeEventListener('keydown', handleKeydown);
+	};
+
+	const addEvents = () => {
+		console.log('adding events');
+		scope.addEventListener('mouseenter', handleMouseenter);
+		scope.addEventListener('mouseleave', handleMouseleave);
+		document.addEventListener('click', handleClick);
+		document.addEventListener('touchstart', handleClick);
+		document.addEventListener('keydown', handleKeydown);
+	};
+
+	return { addEvents, removeEvents };
+}
+
+function addDrawerToggles (headerEl, drawerEl) {
+	const controls = [];
+	controls.push(...headerEl.querySelectorAll(`[aria-controls="${drawerEl.id}"]`));
+	controls.push(...drawerEl.querySelectorAll(`[aria-controls="${drawerEl.id}"]`));
+
+	let handleClose;
+	function toggleCallback(state) {
+		if (state === 'close') {
+			handleClose.removeEvents();
+		} else {
+			// don't capture the initial click or accidental double taps etc.
+			// we could use transitionend but scoping is tricky and it needs prefixing and...
+			setTimeout(handleClose.addEvents, LISTEN_DELAY);
+		}
+
+		drawerEl.classList.toggle('o-header__drawer--closing', state === 'close');
+		drawerEl.classList.toggle('o-header__drawer--opening', state === 'open');
+	}
+
+	controls.forEach((control) => {
+		const drawerToggle = new Toggle(control, {
+			target: drawerEl,
+			callback: toggleCallback
+		});
+
+		// Both toggles have the same target, so the toggle function will be the same
+		// If there's a separate handleClose instance for each toggle, removeEvents doesn't work
+		// when the close toggle is clicked
+		if (!handleClose) {
+			handleClose = handleCloseEvents(drawerEl, drawerToggle.toggle);
+		}
+	});
+}
+
+function addSubmenuToggles (drawerEl) {
+	const submenus = drawerEl.querySelectorAll('[id^="o-header-drawer-child-"]');
+
+	Array.from(submenus).forEach(submenu => {
+		const button = drawerEl.querySelector(`[aria-controls="${submenu.id}"]`);
+
+		submenu.setAttribute('aria-hidden', 'true');
+
+		new Toggle(button, {
+			target: submenu,
+			callback: (state) => {
+				button.textContent = button.textContent.replace(/fewer|more/, state === 'open' ? 'fewer' : 'more');
+			}
+		});
+	});
+}
+
+function appendUserDetail (target) {
+	return fetch('https://session-next.ft.com/', {
```

Comment:
Maybe speak to @commuterjoy to see if membership plan on creating and supporting a cors + jsonp session service for ft in general. session-next is a temporary thing which has just ended up living quite a long time. It might be fine for origami to use for now, but longer term it should be membership/origami's responsibility to provide this service